src/test/unit_tests/test_s3select.js

```diff
@@ -0,0 +1,194 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const mocha = require('mocha');
+const assert = require('assert');
+const s3select_utils = require('../../util/s3select');
+const stream = require('stream');
+const nb_native = require('../../util/nb_native')();
+const { Transform } = require('readable-stream');
+const stream_utils = require('../../util/stream_utils');
+
+const csv_small_str =
+`a,b,c,d,e,
+0,414,339,155,67,
+1,60,741,755,698,
+2,572,53,375,241,
+3,966,233,168,371,
+4,858,55,964,901,
+5,630,524,728,489,
+6,355,260,504,32,
+7,636,736,662,50,
+8,75,169,470,487,
+9,910,225,185,834,
+`;
+
+const json_str =
+`[
+{"doublef": 3.3, "intf" : 3, "stringf": "i am a string"},
+{"doublef": 3.3, "intf" : 123, "stringf": "i am a string"}
+]
+`;
+
+class S3SelectStream extends Transform {
```

Comment:
Why aren't you using the `S3SelectStream` in `util/s3select`?

Reply:
Added a comment to explain rationale of the new class.